### ISSUE TITLE
imapfilter fails to log into gmail.com account if password contains a backslash

### ISSUE THREAD

**Issue by rhaas80** [lefcha/imapfilter#184]
Hello all,

imapfilter fails to log into a gmail account if the password contains a backslash. This is my account settings:
```lua
google = IMAP {
    server = 'imap.gmail.com',
    port = 993,
    username = 'rhaas80@gmail.com',
    password = '\\paswordstring',
    ssl = 'auto',
}
```
and this the content of the debug file written:
```
S (5): * OK Gimap ready for requests from 141.142.41.4 k12-v6mb700267819ivn
C (5): 1000 NOOP
S (5): 1000 OK Nothing Accomplished. k12-v6mb700267819ivn
C (5): 1001 CAPABILITY
S (5): 1001 OK Thats all she wrote! k12-v6mb700267819ivn
C (5): 1002 LOGIN "rhaas80@gmail.com" *
S (5): 1002 BAD Failed to parse your command k12-v6mb700267819ivn
imapfilter: IMAP (5): 1002 BAD Failed to parse your command k12-v6mb700267819ivn
C (5): 1003 CAPABILITY
S (5): 1003 OK Thats all she wrote! k12-v6mb700267819ivn
C (5): 1004 NAMESPACE
S (5): 1004 BAD Unknown command k12-v6mb700267819ivn
imapfilter: IMAP (5): 1004 BAD Unknown command k12-v6mb700267819ivn
imapfilter: authentication of rhaas80@gmail.com@imap.gmail.com failed.
stack traceback:
        [C]: in ?
        [C]: in function 'error'
        /usr/share/imapfilter/account.lua:99: in function '_login_user'
        /usr/share/imapfilter/account.lua:59: in function 'IMAP'
```
This works if I escape the backslash in the password string, ie:
```lua
google = IMAP {
    server = 'imap.gmail.com',
    port = 993,
    username = 'rhaas80@gmail.com',
    password = '\\\\paswordstring',
    ssl = 'auto',
}
```
This happens with imapfilter on git commit 22bb48ef030d3db516b7eeeae283f82d2933d156

**Comment by scottkosty** [comment]
The following might be relevant, from `man imapfilter_config`:

>Note that due to Lua using backslash ‘\’ as an escape character for its strings, one has to double backslashes in order to insert a single backslash inside a regular expression pattern:

**Comment by lefcha** [comment]
@scottkosty is correct, you have to double backslash each backslash due to Lua, so you need 4 backslashes in this case... Possibly this note should be at the beginning of the `imapfilter_config(5)` man page, too.

**Comment by lefcha** [comment]
Closed by 28f092e3d6c5a42fe38b02afb80f508fb0392155.
